From 40082874451d495805f79e38ea729f9a124adeec Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 5 Dec 2024 16:03:23 +0800
Subject: [PATCH 184/274] AOSCOS: MIPS: lemote3a-laptop: mark as a bool and
 depend on DRM_RADEON

This driver, since commit "AOSCOS: MIPS: Loongson: Add a hook to turn
on/off DPMS when Fn+F7 is pressed", lemote3a-laptop uses functions from
radeon which are not exported.

Add a dependency between LEMOTE3A_LAPTOP and DRM_RADEON and mark the
former as bool so that the functions would be visible when enabled.

Fixes: "AOSCOS: MIPS: Loongson: Add a hook to turn on/off DPMS when Fn+F7 is pressed"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/platform/mips/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/platform/mips/Kconfig b/drivers/platform/mips/Kconfig
index 7ef9a47556d5..3cad67bfb7b6 100644
--- a/drivers/platform/mips/Kconfig
+++ b/drivers/platform/mips/Kconfig
@@ -38,7 +38,7 @@ config LS2K_RESET
 	  Loongson-2K1000 Reset Controller driver.
 
 config LEMOTE3A_LAPTOP
-	tristate "Lemote Loongson-3A Laptop Driver"
+	bool "Lemote Loongson-3A Laptop Driver"
 	depends on MACH_LOONGSON64
 	select BACKLIGHT_LCD_SUPPORT
 	select LCD_CLASS_DEVICE
@@ -48,7 +48,7 @@ config LEMOTE3A_LAPTOP
 	select VIDEO_OUTPUT_CONTROL
 	select INPUT_SPARSEKMAP
 	select INPUT_EVDEV
-	depends on INPUT
+	depends on INPUT && DRM_RADEON
 	default y
 	help
 	  Lemote Loongson-3A/2Gq family laptops driver.
-- 
2.47.1

